regexes/oss.yml

```diff
@@ -254,17 +254,49 @@
   name: 'Fire OS'
   version: '$1'
 
-- regex: '.+FIRETVSTICK2018'
+- regex: 'Andr[o0]id 2.3.3.+ AFT'
+  name: 'Fire OS'
+  version: '1'
+
+- regex: 'Andr[o0]id 4.0.+ AFT'
+  name: 'Fire OS'
+  version: '2'
+
+- regex: 'Andr[o0]id 4.0.[34].+ KF[ADFGJKMORSTQ]'
+  name: 'Fire OS'
+  version: '3'
+
+- regex: 'Andr[o0]id 4.2.2.+ (?:AFT|KF[ADFGJKMORSTQ])'
+  name: 'Fire OS'
+  version: '3'
+
+- regex: 'Andr[o0]id 4.4.2.+ AFT'
+  name: 'Fire OS'
+  version: '4'
+
+- regex: 'Andr[o0]id 4.4.3.+ (?:AFT|KF[ADFGJKMORSTQ])'
+  name: 'Fire OS'
+  version: '4.5.1'
+
+- regex: 'Andr[o0]id 5.1.+ (?:AFT|KF[ADFGJKMORSTQ])'
+  name: 'Fire OS'
+  version: '5'
+
+- regex: 'Andr[o0]id 7.1.2.+ (?:AFT|KF[ADFGJKMORSTQ])|.+FIRETVSTICK2018'
   name: 'Fire OS'
   version: '6'
 
-- regex: 'AFTSO001'
+- regex: 'Andr[o0]id 9.+ (?:AFT|KF[ADFGJKMORSTQ])|AFTSO001'
   name: 'Fire OS'
   version: '7'
 
-- regex: 'Andr0id 9.+AFTdev'
+- regex: 'Andr[o0]id 1[01].+ (?:AFT|KF[ADFGJKMORSTQ])'
   name: 'Fire OS'
-  version: '7'
+  version: '8'
```

Comment:
I suggest combining these rules as we did with iOS

```yml
- regex: 'Andr[o0]id (\d([\d.])*);? AFT|KF[ADFGJKMORSTQ]|.*FIRETVSTICK2018'
  name: 'Fire OS'
  versions:
       - regex: 'Andr[o0]id 1[01].+ (?:AFT|KF[ADFGJKMORSTQ])'
         version: '8'
       - regex: 'Andr[o0]id 9.+ (?:AFT|KF[ADFGJKMORSTQ])|AFTSO001'
         version '7'  
      - regex: 'Andr[o0]id 7|.+FIRETVSTICK2018'
        version: '6'
      - regex: 'Andr[o0]id 5.1'
        version: '5'
      - regex: 'Andr[o0]id 4.0.[34]'
        version: '3'
      - regex: 'Andr[o0]id 4.2.2'
        version: '3'
      - regex: 'Andr[o0]id 4.4.2'
        version: '4'
     - regex: 'Andr[o0]id 4.4.3'
       version: '4.5.1'
    - regex: 'Andr[o0]id 2.3.3'
        version: '1'
      - regex: 'Andr[o0]id 4.0'
        version: '2'
```

Reply:
I've improved the regex a little bit, because there were some devices in podcasting.yml without OS.